sympy/series/fourier.py

```diff
@@ -440,16 +438,210 @@ def __sub__(self, other):
+        if not (type(exprs) == tuple and len(exprs) == 3):  # exprs is not of form (a0, an, bn)
+            # Converts the expression to fourier form
+            c, e = exprs.as_coeff_add()
+            rexpr = c + Add(*[TR10(i) for i in e])
+            a0, exp_ls = rexpr.expand(trig=False, power_base=False, power_exp=False, log=False).as_coeff_add()
+
+            x = limits[0]
+            L = abs(limits[2] - limits[1]) / 2
+
+            a = Wild('a', properties=[lambda k: k.is_Integer, lambda k: k is not S.Zero, ])
+            b = Wild('b', properties=[lambda k: x not in k.free_symbols, ])
+
+            an = dict()
+            bn = dict()
+
+            # separates the coefficients of sin and cos terms in dictionaries an, and bn
+            for p in exp_ls:
+                t = p.match(b * cos(a * (pi / L) * x))
+                q = p.match(b * sin(a * (pi / L) * x))
+                if t:
+                    an[t[a]] = t[b] + an.get(t[a], S.Zero)
+                elif q:
+                    bn[q[a]] = q[b] + bn.get(q[a], S.Zero)
+                else:
+                    a0 += p
+
+            exprs = (a0, an, bn)
+
+        args = map(sympify, (f, limits, exprs))
+
+        return Expr.__new__(cls, *args)
+
+    @property
+    def interval(self):
+        _length = 1 if self.a0 else 0
+        _length += max(set(self.an.keys()).union(set(self.bn.keys()))) + 1
+        return Interval(0, _length)
+
+    @property
+    def length(self):
+        return self.stop - self.start
+
+    @property
+    def L(self):
+        return abs(self.period[1] - self.period[0]) / 2
+
+    def truncate(self, n=oo):
+        """
+        Return the first n nonzero terms of the series if n is less than the size of the finite fourier series.
+
+        If n is None return an iterator.
+
+        See Also
+        ========
+
+        sympy.series.fourier.FourierSeries.truncate
+        """
+        if n is None:
+            return iter(self)
+
+        terms = []
+        for t in self:
+            if len(terms) == n:
+                break
+            if t is not S.Zero:
+                terms.append(t)
+
+        return Add(*terms)
 
+    def shiftx(self, s):
+        """Shift x by a term independent of x.
+
+        f(x) -> f(x + s)
 
-def fourier_series(f, limits=None):
+        This is fast, if Fourier series of f(x) is already
+        computed.
+
+        See Also
+        ========
+        sympy.series.fourier.FourierSeries.shiftx
+
+        """
+        s, x = sympify(s), self.x
+
+        if x in s.free_symbols:
+            raise ValueError("'%s' should be independent of %s" % (s, x))
+
+        _expr = self.truncate().subs(x, x + s)
+        sfunc = self.function.subs(x, x + s)
+
+        return self.func(sfunc, self.args[1], _expr)
+
+    def scale(self, s):
+        """Scale the function by a term independent of x.
+
+        f(x) -> s * f(x)
+
+        This is fast, if Fourier series of f(x) is already
+        computed.
```

Comment:
What does "already computed" mean in this context? Aren't finite fourier series always computed in the constructor?

It's not necessary to duplicate the docstring here if it will be the same as the superclass. `help()` and IPython's `?` automatically look at the superclass method's docstring if a method doesn't have a docstring of its own.

Reply:
Nothing is computed in constructor just the form of expression is changed. The expression is "already" fourier series and the sense with this description is that it is more feasible to have `s*f(x)` when fourier series of `f(x)` is found. But since it appears to be a duplication of docstring I think I should remove these docstrings.